subprojects/core/src/main/java/org/gradle/process/internal/streams/ExecOutputHandleRunner.java

```diff
@@ -62,7 +63,7 @@ public void run() {
     }
 
     public void run(Executor executor) {
-        executor.execute(this);
+        executor.execute(new BuildOperationIdentifierPreservingRunnable(this));
```

Comment:
Ah, based on what I just wrote... does it make any sense that we wrap the `standardInputRunner` with a build operation?